backend/coreapp/tests.py

```diff
@@ -39,7 +38,43 @@ def test_accept_late_rodata(self):
         self.assertEqual(response.status_code, status.HTTP_201_CREATED)
         self.assertEqual(Scratch.objects.count(), 1)
 
-class CompilationTests(TestCase):
+class CompilationTests(APITestCase):
+    scratch_url: str
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.scratch_url = reverse('scratch')
+
+    def test_simple_compilation(self):
+        """
+        Ensure that we can run a simple compilation via the api
+        """
+        scratch_dict = {
+            'arch': 'mips',
+            'context': '',
+            'target_asm': 'glabel func_80929D04\njr $ra\nnop'
+        }
+
+        # Test that we can create a scratch
+        response = self.client.post(self.scratch_url, scratch_dict)
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(Scratch.objects.count(), 1)
+
+        slug = response.json()["scratch"]["slug"]
+
+        compile_dict = {
+            'slug': slug,
+            'compiler': 'gcc2.8.1',
+            'cc_opts': '-mips2 -O2',
+            'source_code': 'int add(int a, int b){\nreturn a + b;\n}\n'
+        }
+
+        # Test that we can compile a scratch
+        response = self.client.post(f"{self.scratch_url}/{slug}/compile", compile_dict)
```

Comment:
`reverse("compile_scratch", kwargs={"slug": slug})`; don't construct URLs manually if you can avoid it